CMakeLists.txt

```diff
@@ -36,9 +61,28 @@ if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" STRE
     # set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -D_GLIBCXX_DEBUG")
 elseif (MSVC)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4244 /wd4250 /wd4251 /wd4267 /wd4275")
+    if (NOT BUILD_SHARED_LIBS AND BUILD_STATIC_LIBS)
+        if (MSVC_STATIC_RUNTIME_LIBRARY)
+            set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+        endif ()
+    else()
+        if (MSVC_STATIC_RUNTIME_LIBRARY)
+            message(WARNING "MSVC_STATIC_RUNTIME_LIBRARY should be used with BUILD_SHARED_LIBS=OFF and BUILD_STATIC_LIBS=ON.")
+        endif ()
+    endif ()
 endif ()
 
-set(Boost_USE_STATIC_LIBS OFF)
+if (BOOST_STATIC_LIBS)
+    set(Boost_USE_STATIC_LIBS ON)
+    if (MSVC_STATIC_RUNTIME_LIBRARY)
+        set(Boost_USE_STATIC_RUNTIME ON)
+    else ()
+        set(Boost_USE_STATIC_RUNTIME OFF)
+    endif ()
+else ()
+    set(Boost_USE_STATIC_LIBS OFF)
+    set(Boost_USE_STATIC_RUNTIME OFF)
+endif ()
```

Comment:
I propose to set `Boost_USE_STATIC_RUNTIME` according to the value of `MSVC_STATIC_RUNTIME_LIBRARY` even if `BOOST_STATIC_LIBS` is not set.
```
set(Boost_USE_STATIC_LIBS ${BOOST_STATIC_LIBS})
set(Boost_USE_STATIC_RUNTIME ${MSVC_STATIC_RUNTIME_LIBRARY})
```

Reply:
done